0_From_Code_First_To_First_Principles/1_code_first/assignment/src/controller/CreateUser/ICreateUserRequest.ts

```diff
@@ -0,0 +1,20 @@
+export interface ICreateUserRequest {
```

Comment:
Hey brother! Great work so far on this - sorry it took me a while to get to your submission here, I was just deep in recording. 

I like how you've broken things up into command objects. 

What are your thoughts on _not using the "I" prefix_ for your interfaces? 

I recorded a video [earlier about this](https://www.essentialist.dev/products/the-software-essentialist/categories/2152815724/posts/2167861671), but the main idea is just that _interfaces_, _abstract classes_ and _types_ still refer to real things, so we should try to avoid using the _I_ prefix because it has a strange downstream effect of forcing us to not really know what it is we're referring to.

My recommendation: Instead of "I"-prefixing your types/structural data, use the "Props" word.

I'd call this CreateUserRequestProps or CreateUserRequestData.

What do you think?

Reply:
Hey Khalil! Thanks for taking you time on reviewing my code. I've doubt all the time on naming the interfaces, the practice of naming with the "I" prefix it's widely extended and leads me to doubt what is the correct choice, but i'm totally agree with you. There is a spanish developer widely known in the community called **Carlos Ble** who says that "I" on the interfaces just creates unnecessary noise, the IDE itself give us the information of what really the element is. I'll try to stick with this.  Ty!